### DISCUSSION TITLE
pac modelbuilder - support javascript language

### DISCUSSION THREAD

**Discussion by kalluu-ck** [microsoft/powerplatform-vscode#613] [Ideas]
Dear team,

In my project, we work a lot with Web API, and now we don't want to hardcode the logical / schema names of entity fields.
So is there a way of using the `pac modelbuilder --language=javascript` to generate js files with all the needed metadata (schema names, logical names of entity, fields and relationships) of an entity? Then we can just import and use them instead of hardcoding.

Thanks,
Kim Anh

**Comment by rajyraman** [comment]
@kalluu-ck - I am using [dataverse-ify](https://github.com/scottdurow/dataverse-ify) along with TypeScript for this.

**Comment by tehcrashxor** [comment]
The DataverseModelBuilder library used by PAC now has its own GitHub repo - https://github.com/microsoft/PowerPlatform-DataverseModelBuilder
